src/services/ghost/strategies/UserRequestStrategy.ts

```diff
@@ -0,0 +1,165 @@
+			prompt += `- Total Lines: ${lineCount}\n\n`
+		}
+
+		// If there's a selection, highlight it
+		if (context.range && !context.range.isEmpty && context.document) {
+			prompt += this.formatSelectedText(context.document, context.range)
+			prompt += "\n"
+			prompt += "**Note**: The user has selected this specific code. "
+			prompt += "Focus your changes on this selection unless the request requires changes elsewhere.\n\n"
+		}
+
+		// Include diagnostics if present (user might be asking to fix them)
+		if (context.diagnostics && context.diagnostics.length > 0) {
+			prompt += this.formatDiagnostics(context.diagnostics)
+			prompt += "\n"
+		}
+
+		// Add AST context if available
+		if (context.rangeASTNode) {
+			prompt += this.formatASTContext(context.rangeASTNode)
+			prompt += "\n"
+		}
+
+		// Add the full document with cursor marker
+		if (context.document) {
+			prompt += "## Full Code\n"
+			prompt += this.formatDocumentWithCursor(context.document, context.range)
+			prompt += "\n\n"
+		}
+
+		// Add specific instructions
+		prompt += "## Instructions\n"
+		prompt += `Generate changes that directly implement: "${context.userInput}"\n`
+
+		if (context.range && context.document) {
+			prompt += `Start with changes at the cursor position (${this.CURSOR_MARKER}).\n`
+
+			// Add context-specific hints
+			if (context.range.isEmpty) {
+				prompt += "The cursor is at a specific position - consider adding new code there if appropriate.\n"
+			} else {
+				prompt += "The user has selected code - focus on modifying or replacing the selection.\n"
+			}
+		}
+
+		// Add hints based on common request patterns
+		const request = context.userInput?.toLowerCase() || ""
+		if (request.includes("fix")) {
+			prompt += "Look for errors, warnings, or issues that need to be resolved.\n"
+		} else if (request.includes("add") || request.includes("implement")) {
+			prompt += "Add new functionality as requested, ensuring it integrates well with existing code.\n"
+		} else if (request.includes("refactor") || request.includes("improve")) {
+			prompt += "Improve code structure and readability while maintaining the same functionality.\n"
+		} else if (request.includes("test")) {
+			prompt += "Generate appropriate test cases or test-related code.\n"
+		} else if (request.includes("comment") || request.includes("document")) {
+			prompt += "Add clear, helpful comments or documentation.\n"
+		}
+
+		return prompt
+	}
+
+	/**
+	 * Override to provide more context for certain types of requests
+	 */
+	override getUserPrompt(context: GhostSuggestionContext): string {
+		// For certain requests, we might want to include more context
+		const request = context.userInput?.toLowerCase() || ""
+
+		// If the request mentions other files or imports, include more context
+		if (request.includes("import") || request.includes("from")) {
+			// Could potentially include information about available modules
+			// For now, we'll use the standard approach
+		}
```

Comment:
Ah yeah I think this type of thing would be the next thing we'd work on if we had more time